pkg/ingester/instance.go

```diff
@@ -89,10 +93,16 @@ func (i *instance) consumeChunk(ctx context.Context, labels []client.LabelAdapte
 		i.index.Add(labels, fp)
 		i.streams[fp] = stream
 		i.streamsCreatedTotal.Inc()
+		memoryStreams.Inc()
 		i.addTailersToNewStream(stream)
 	}
 
-	return stream.consumeChunk(ctx, chunk)
+	err := stream.consumeChunk(ctx, chunk)
+	if err != nil {
+		memoryChunks.Inc()
+	}
```

Comment:
Is this right? i.e., only increase the number of memory chunks if there was an error?

Reply:
Nope.  Good catch.  Fixed.